### ISSUE TITLE
How to rank search results based on boolean fields in Typesense?

### ISSUE THREAD

**Issue by tusharmalpani20** [typesense/typesense#999]
## Description

I have a Typesense collection with fields like **is_pricing_available** and **is_demo_available** which are **boolean** values. I would like to boost the ranking of documents where these fields are true.

## Expected Behavior

I expect that the search results will be ordered such that documents where **is_pricing_available** and **is_demo_available** are true will appear higher in the ranking.

## Actual Behavior

Currently, the search results do not seem to be ordered based on the values of these boolean fields.

## Metadata

**Typesense Version**: 0.24.1

**OS**: Typesense Cloud

**Comment by jasonbosco** [comment]
You want to use the Optional Filtering feature described here: https://typesense.org/docs/0.24.1/api/search.html#sorting-based-on-conditions

**Comment by tusharmalpani20** [comment]
> You want to use the Optional Filtering feature described here: https://typesense.org/docs/0.24.1/api/search.html#sorting-based-on-conditions

Thank you for your suggestion of using the Condition based sorting feature to boost the ranking of documents based on boolean fields in Typesense. However, I am looking for a more nuanced approach to boost the ranking of documents based on multiple boolean fields and key_match-based scores rather than just one. I am interested in using a scoring-based mechanism that takes into account all the boolean fields and the matched key score that I want to use for ranking, and assigns weights to each field based on its importance.

I appreciate your help so far and I am looking forward to hearing a solution that can help me achieve this. If you have any further suggestions or resources that could guide me on how to use Custom Ranking to achieve this, I would greatly appreciate it.

Thank you!

**Comment by jasonbosco** [comment]
You can set field priorities using the `query_by_weights` search parameter. 

And then to boost based on two bool fields, you can do something like this:

```
sort_by: _eval(is_pricing_available:true && is_demo_available:true):desc
```

This will prioritize all items that match that condition first, followed by the other items. 

If that isn't what you're looking for, could you give me a concrete example with a few sample documents, a sample query and how you want the ranking to be?

**Comment by tusharmalpani20** [comment]
Hey @jasonbosco ,

I appreciate your suggestion, but I'm looking to implement a more nuanced scoring mechanism. Setting the fields is_pricing_available and is_demo_available to true would boost the ranking of all documents with those values, regardless of their overall relevance to the user's query.

Instead, I'd like to prioritize documents based on a combination of their key matched score and the boolean fields. For example, let's say we have two documents - one with is_pricing_available and is_demo_available set to true but a low key matched score, and another with those fields set to false but a high key matched score. In this case, I'd want the second document to be ranked higher, as it's more relevant to the user's query despite not having those boolean fields set to true.

My goal is to ensure that the most relevant documents are shown first, while still taking into account the impact of those boolean fields on ranking.

**Comment by jasonbosco** [comment]
> Setting the fields is_pricing_available and is_demo_available to true would boost the ranking of all documents with those values, regardless of their overall relevance to the user's query.

No, Typesense will first execute the search query, pick the set of relevant documents and then apply the boolean based sorting logic on top of those results.

**Comment by sandstrom** [comment]
Would this be solved by this one? https://github.com/typesense/typesense/issues/376
